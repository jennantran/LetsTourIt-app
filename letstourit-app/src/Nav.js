import React, { Component } from 'react';
import './Nav.css';

class Nav extends Component {
    render(){
        return(
            <section className='nav'>
                <h2>JustTourIt.</h2>
                <ul>
                    <li><a href='/'>Home</a></li>
                    <li><a href='/logIn'>Log in </a></li>
                    <li><a href='/favoritedList'>Favorited List</a></li>
                </ul>
            </section>
        );
    }
}

export default Nav; 