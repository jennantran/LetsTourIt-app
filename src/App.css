* {
    box-sizing: border-box;
  }

  header{
    display: block;
  }

  
  body {
    font-family: 'Alegreya';
    margin: 0;
    padding: 0;
    color: #444;
    line-height: 1.25;
    display: inline;
    font-size: 18px;
  }

  nav,
  footer {
    padding: .10% 0;
    height: 45px;
    background-color: #68d9a9;
  }

  footer {
    position:fixed;
    left:0;
    bottom:0;
    right:0;
    font-size: 15px;
  }

  nav {
    position: fixed;
    top: 0;
    left: 0;
    right: 0;
  }

  ul.nav{
    text-decoration: none;
  }

  h2{
    font-size: 16px;
    display:flex;
  }

  .compass{
    width: 200px;
    height: 400px;
    padding-top: 50px; 
    margin-left: 20px;
}
  img.compass{
    margin: 50px;
} 
  .header{
    padding: 150px;
    margin-top: 50px;
    width: 500px;
}


@media screen and (min-width:600px) {
    
  h2{
    font-size: 20px;
  }

  .header{
    width: 1000px;
  }
}
