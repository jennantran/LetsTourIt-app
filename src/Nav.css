.nav{
    display: flex;
    flex-direction: row;
    justify-content: center;
    align-items: center;
}
h2{
    left: 15px;
}
ul{
    list-style-type: none;
}

li{
    display:inline;
}

a{
    text-decoration: none;
    margin-right: 15px;
    color: white;
    text-shadow: -1px ;
  
}
a:hover {
    text-decoration: underline;
  }