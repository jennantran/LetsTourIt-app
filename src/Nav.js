import React, { Component } from 'react';
import './Nav.css';

class Nav extends Component {
    render(){
        return(
            <section className='nav'>
                <h2>JustTourIt.</h2>
                <ul> 
                    <li><a href='/'>Home</a></li>
                    <li><a href='/login'>Log in </a></li>
                    <li><a href='/search'>Search </a></li>
                    <li><a href='/favorites'>Favorites</a></li>
                </ul>
            </section>
        );
    }
}

export default Nav; 