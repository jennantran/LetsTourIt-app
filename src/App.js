import React, { Component } from 'react';
import { Route } from 'react-router-dom';
import HomePage from './HomePage/HomePage';
import Nav from './Nav';
import Footer from './Footer';
import SearchPage from './SearchPage';
import Login from './auth/Login';
import './App.css';
import FavoritedList from './FavoritedList';
import FavoritesContext from './FavoritesContext';
import dummyStore from './dummyStore';
import SignUp from './auth/SignUp';

class App extends Component {
   state = {
     favorites: [],
     addFavorite: this.addFavorite,
     deleteFavorite: this.deleteFavorite
   }
  
   componentDidMount() {
    // fake date loading from API call
    setTimeout(() => this.setState(dummyStore), 600);
}

   addFavorite = favorite => {
     this.setState({
       favorites: [...this.state.favorites,favorite]
     })
     setTimeout(() => console.log(this.state));
     
   }

  deleteFavorite = (faveId) =>{
    console.log(faveId);
    console.log(this);
    this.setState({
      favorites: this.state.favorites.filter(favorite => favorite.id !== faveId)
    })
    setTimeout(() => console.log(this.state));
  }

  render(){
    const contextValue = {
      favorites: this.state.favorites,
      addFavorite: this.addFavorite,
      deleteFavorite: this.deleteFavorite
    }
    console.log(this.state);
    return (
      <div className='app'>
      <FavoritesContext.Provider value={contextValue}>
        <nav>
          <Nav/>
        </nav>
        <main>
          <Route exact path='/' component={HomePage} />
          <Route path='/search' component={SearchPage} />
          <Route path='/login' component={Login} />
          <Route path='/signUp' component={SignUp} />
          <Route path='/favorites' component={FavoritedList} />
    
        </main>
        </FavoritesContext.Provider>
        <footer>
          <Footer/>
        </footer>
      </div>
    );
  }
}

export default App;