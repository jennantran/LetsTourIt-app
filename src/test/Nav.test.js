import React from 'react';
import Nav from '../Nav';
import { BrowserRouter } from 'react-router-dom';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import App from  '../App';

it('renders Nav component without crashing', () => {
  const div = document.createElement("div");
  ReactDOM.render(
    <BrowserRouter>
      <App>
          <Nav/>
      </App>
    </BrowserRouter>,
    div
  );
  //clean up code
  ReactDOM.unmountComponentAtNode(div);
});

it('renders the Nav UI as expected', () => {
    const tree = renderer
      .create(
        <BrowserRouter>
        <App>
           <Nav/>
        </App>
      </BrowserRouter>,
      )
      .toJSON();
  
    expect(tree).toMatchSnapshot();
  });