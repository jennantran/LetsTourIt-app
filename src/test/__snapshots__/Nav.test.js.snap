// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the Nav UI as expected 1`] = `
<div
  className="app"
>
  <nav>
    <section
      className="nav"
    >
      <h2>
        JustTourIt.
      </h2>
      <ul>
        <li>
          <a
            href="/"
            onClick={[Function]}
          >
            Home
          </a>
        </li>
        <li>
          <a
            href="/search"
            onClick={[Function]}
          >
            Search 
          </a>
        </li>
        <li>
          <a
            href="/favorites"
            onClick={[Function]}
          >
            Favorites
          </a>
        </li>
        <li
          className="notLogged-in"
        >
          <a
            href="/login"
            onClick={[Function]}
          >
            Log in
          </a>
        </li>
      </ul>
    </section>
  </nav>
  <main>
    <div
      className="homePage"
    >
      <section
        className="homePageSect"
      >
        <div>
          <img
            alt="bay-bridge"
            className="compass"
            src="compass.jpg"
          />
        </div>
        <div
          className="header"
        >
          <h1>
            Let the Exploring Begin
          </h1>
          <p>
            <em>
              Don't know what to do? 
            </em>
          </p>
          <p>
            <em>
              Search for things to do and things to eat. 
            </em>
          </p>
          <button
            onClick={[Function]}
          >
            Sign Up
          </button>
        </div>
      </section>
    </div>
  </main>
  <footer>
    <footer
      className="Footer"
    >
      <p>
        Â©Jennifer Tran, 2020. All Rights Reserved.
      </p>
    </footer>
  </footer>
</div>
`;
